Product.cs

```diff
@@ -0,0 +1,40 @@
+﻿public class Product
+{
+    private decimal _price;
+    private int _quantity;
+    public string Name { get; set; }
+
+    public Product(string name, decimal price, int quantity) 
+    {
+        (Name, Price, Quantity) = (name, price, quantity);
+    }
+
+    public decimal Price
+    {
+        get { return _price; }
+        set 
+        {
+            if (value < 0)
+                Console.WriteLine("Please try again and enter a positive price.");
```

Comment:
I don't think it's this property responsibility to validate the value, you already have another method `AreValid` that validates the values, so you are repeating the logic, and that violates the `DRY` principle. Lets have that `ProductValidator` I mentioned above, so you will be validating the values only once before adding the products, so no need to do validations again here.